src/hotspot/share/code/nmethod.cpp

```diff
@@ -1160,6 +1160,8 @@ void nmethod::finalize_relocations() {
 
 void nmethod::make_deoptimized() {
   if (!Continuations::enabled()) {
+    // Don't deopt this again.
+    set_deoptimized_done();
     return;
   }
```

Comment:
I don't understand this check-and-bailout logic at all. If continuations are
not enabled, then why does `nmethod::make_deoptimized()` do nothing?
This isn't an issue introduced by your patch, but I just happened to notice
when I was reading thru...

Reply:
We two deopts mechanisms:
* Patching return PC into frames for nmethod that are no longer valid.
   Done by handshaking all threads and walking their stack.
   
* Virtual threads stacks are only patched this way when mounted.
   If we have a million VT we can't walk the entire heap and patch PC is all these stack.
   Instead there is nop-sledge after each calls from nmethods, so patch these to jump to the deopt
   when returning into a deopted nmethod.
   Calls into VM (causing the VT to pin) do not have these nop-sledges, so when using Loom, both mechanism is needed.

Reply:
Thanks for explanation.